src/gui/FxMixerView.cpp

```diff
@@ -439,38 +439,28 @@ void FxMixerView::deleteUnusedChannels()
 	tracks += Engine::getSong()->tracks();
 	tracks += Engine::getBBTrackContainer()->tracks();
 
-	// go through all FX Channels
-	for(int i = m_fxChannelViews.size()-1; i > 0; --i)
+	bool inUse[m_fxChannelViews.size()] = {false};
+
+	for (Track* t: tracks)
 	{
-		// check if an instrument references to the current channel
-		bool empty=true;
-		for( Track* t : tracks )
+		int channel = 0;
```

Comment:
```suggestion
		//The channel that this track sends to. Since master channel is always in use,
		//setting this to 0 is a safe default (for tracks that don't sent to the mixer).
		int channel = 0;
```